src/app/signup/school-admin/page.tsx

```diff
@@ -0,0 +1,102 @@
+'use client';
+import { useState } from 'react';
+import signUp from '@/firebase/auth/signUp';
+import { doc, setDoc } from 'firebase/firestore';
+import { getFirestore } from 'firebase/firestore';
+
+const SchoolAdminSignup = () => {
+    const [name, setName] = useState('');
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [confirmPassword, setConfirmPassword] = useState('');
+    const [error, setError] = useState('');
+    const [success, setSuccess] = useState<boolean>(false);
+
+    const handleSubmit = async (event: React.FormEvent) => {
+        event.preventDefault();
+
+        // Check if passwords match
+        if (password !== confirmPassword) {
+            setError('Passwords do not match.');
+            return;
+        }
+
+        const { result, error } = await signUp(email, password);
+        if (error) {
+            setError(error.message);
+        } else {
+            try {
+                const db = getFirestore();
```

Comment:
we'll probably have extract this firebase update into a server component or route handler in the future :D but dw about it for now!!